src/Symfony/Component/Workflow/Registry.php

```diff
@@ -24,17 +23,11 @@ class Registry
     private $workflows = array();
 
     /**
-     * @param Workflow                        $workflow
-     * @param string|SupportStrategyInterface $supportStrategy
+     * @param Workflow                 $workflow
+     * @param SupportStrategyInterface $supportStrategy
      */
-    public function add(Workflow $workflow, $supportStrategy)
+    public function add(Workflow $workflow, SupportStrategyInterface $supportStrategy)
```

Comment:
@Pierstoval the issue is that changing the typehint does not allow them to support both 3.4 and 4.0 at the same time in their child class => no continuous upgrade path)